src/proxy-ui-api/frontend/src/store/modules/addClient.ts

```diff
@@ -143,13 +161,21 @@ export const actions: ActionTree<AddClientState, RootState> = {
     // Fetch clients from backend that can be selected
     return api
       .get(
-        '/clients?exclude_local=true&member_missing_sign_cert=true&internal_search=false&show_members=false',
+        '/clients?exclude_local=true&internal_search=false&show_members=false',
       )
-      .then((res) => {
-        commit('storeSelectableClients', res.data);
+      .then((globalClientsRes) => {
+        // Fetch list of local clients and filter out global clients
+        // that have local relatives
+        api.get('/clients').then((localClientsRes) => {
+            commit('storeSelectableClients',
+            excludeClientsWithLocalRelatives(globalClientsRes.data, localClientsRes.data));
+          })
+          .catch((localClientsError) => {
+            throw localClientsError;
+          });
       })
-      .catch((error) => {
-        throw error;
+      .catch((globalClientsError) => {
+        throw globalClientsError;
       });
```

Comment:
Making an async call inside anothers success-block creates what is called a callback hell. This also introduces an extra period of waiting as the latter call is created after the first has been successfully completed. 

With Promise.all we can send multiple rest calls and wait for ALL of them have completed successfully. The code sample of how this is done is below. I didn't test it but it's at least nearly there...

``` 
// Fetch list of local clients and filter out global clients
    // that have local relatives
    return Promise.all([globalClientsPromise, localClientsPromise])
      .then( (response) => {
        const globalClients = response[0];
        const localClients = response[1];
        commit('storeSelectableClients', excludeClientsWithLocalRelatives(globalClients.data, localClients.data));
      })
      .catch( error => {
        throw error;
      });

```

Reply:
Thanks for the suggestion! I updated the implementation accordingly. It worked almost directly - just to had to define `globalClientsPromise` and `localClientsPromise`.